dynamo-migration-lambda/app/RdsMigration.py

```diff
@@ -0,0 +1,154 @@
+
+AWS_REGION = 'eu-west-2'
+TARGET_ENVIRONMENT = os.environ["ENVIRONMENT_NAME"]
+NEW_TABLE_NAME = f'{TARGET_ENVIRONMENT}-ehr-transfer-tracker'
+
+ssm_client = boto3.client("ssm")
+dynamo_client = boto3.client("dynamodb", region_name=AWS_REGION)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+conversations_to_delete = {}
+
+class MessageRowItem(Enum):
+    INBOUND_CONVERSATION_ID = 0
+    LAYER = 1
+    INBOUND_MESSAGE_ID = 2
+    CREATED_AT = 3
+    UPDATED_AT = 4
+    DELETED_AT = 5
+    PARENT_ID = 6
+
+
+@dataclass
+class DatabaseCredentials:
+    user: str
+    password: str
+    host: str
+    database: str
+
+
+def _get_new_datetime(old_datetime: datetime) -> str:
+    if old_datetime is not None:
+        target_timezone = ZoneInfo('Europe/London')
+        formatted_datetime = str(old_datetime).split('.')[0]
+        parsed_datetime = parser.parse(formatted_datetime).astimezone(target_timezone).isoformat()
+
+        return str(parsed_datetime)
+
+
+def _fetch_ssm_parameter(name: str, decrypt: bool = False) -> str:
+    return ssm_client.get_parameter(Name=name, WithDecryption=decrypt)['Parameter']['Value']
+
+
+def _get_rds_credentials() -> DatabaseCredentials:
+    credentials = {
+        "host": _fetch_ssm_parameter(f'/repo/{TARGET_ENVIRONMENT}/output/prm-deductions-ehr-repository/db-host'),
+        "user": _fetch_ssm_parameter(f'/repo/{TARGET_ENVIRONMENT}/user-input/ehr-repo-db-username', True),
+        "password": _fetch_ssm_parameter(f'/repo/{TARGET_ENVIRONMENT}/user-input/ehr-repo-db-password', True),
+        "database": _fetch_ssm_parameter(f'/repo/{TARGET_ENVIRONMENT}/output/prm-deductions-ehr-repository/db-name', True)
+    }
+
+    return DatabaseCredentials(**credentials)
+
+
+def _get_core_and_fragments(connection) -> list[tuple]:
+    logger.info("Fetching CORE/FRAGMENT(s) from RDS.")
+
+    with connection.cursor() as cursor:
+        statement = """
+        SELECT
+            conversation_id AS InboundConversationId,
+            CASE
+              WHEN type = 'ehrExtract' THEN 'CORE'
+              ELSE concat('FRAGMENT#', message_id)
+            END AS Layer,
+            message_id AS InboundMessageId,
+            created_at AS CreatedAt,
+            updated_at AS UpdatedAt,
+            deleted_at AS DeletedAt,
+            parent_id AS ParentId
+        FROM messages;
+        """
+
+        cursor.execute(statement)
+        result = cursor.fetchall()
+
+        logger.info(f"Found {len(result)} records when querying the RDS database.")
+        return result
+
+
+def _get_dynamo_items(rds_result_set: list[tuple]) -> list[dict]:
+    logger.info(f"Constructing DynamoDB items from RDS result set.")
+    dynamo_items = []
+
+    for row in rds_result_set:
+        inbound_conversation_id = row[MessageRowItem.INBOUND_CONVERSATION_ID.value].upper()
+        inbound_message_id = row[MessageRowItem.INBOUND_MESSAGE_ID.value].upper()
+        layer = row[MessageRowItem.LAYER.value]
```

Comment:
Layer is created on line 77 `concat('FRAGMENT#', message_id)`. It looks like we're not doing anything to upper case this anywhere else in the code from what I can tell. Does the message ID component need uppercasing?

Reply:
I have added .upper() to line 101.